workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/RateLimitedPositionSendSystem.cs

```diff
@@ -0,0 +1,65 @@
+using Improbable.Gdk.Core;
+using Improbable.Transform;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.TransformSynchronization
+{
+    [DisableAutoCreation]
+    [UpdateAfter(typeof(DefaultUpdateLatestTransformSystem))]
+    [UpdateAfter(typeof(RateLimitedTransformSendSystem))]
+    [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
+    public class RateLimitedPositionSendSystem : ComponentSystem
+    {
+        private struct Data
+        {
+            public readonly int Length;
+            public ComponentDataArray<Position.Component> PositionComponent;
+            public ComponentDataArray<LastPositionSentData> LastPositionSent;
+            [ReadOnly] public ComponentDataArray<TransformInternal.Component> TrasnformComponent;
+            [ReadOnly] public SharedComponentDataArray<RateLimitedSendConfig> RateLimitedConfig;
+
+            [ReadOnly] public ComponentDataArray<Authoritative<Position.Component>> DenotesAuthoritative;
+        }
+
+        [Inject] private Data data;
+
+        protected override void OnUpdate()
+        {
+            for (int i = 0; i < data.Length; ++i)
+            {
+                var position = data.PositionComponent[i];
+
+                var lastPositionSent = data.LastPositionSent[i];
+                lastPositionSent.TimeSinceLastUpdate += Time.deltaTime;
+                data.LastPositionSent[i] = lastPositionSent;
+
+                if (lastPositionSent.TimeSinceLastUpdate <
+                    1.0f / data.RateLimitedConfig[i].MaxPositionUpdateRateHz)
+                {
+                    continue;
+                }
+
+                var transform = data.TrasnformComponent[i];
+                var coords = new Coordinates
```

Comment:
Not for this PR - is it worth writing extension methods for conversions?

Reply:
Eventually maybe but I don't like doing this for lossy conversions.